machine-learning/lambda-v2/lambda-artifacts/com.lambda.DLRObjectDetection/1.0.0/object_detection/prediction_utils.py

```diff
@@ -0,0 +1,286 @@
+            + "width=(int)1920, height=(int)1080, format=(string)NV12,"
+            + "framerate=(fraction)30/1 ! nvvidconv flip-method=2 !"
+            + "video/x-raw, width=(int)1920, height=(int)1080,"
+            + "format=(string)BGRx ! videoconvert ! appsink"
+        )
+    elif platform.machine() == "x86_64":  # Deeplens
+        import awscam
+
+        config_utils.CAMERA = awscam
+
+
+def draw_bounding_boxes(original_image, inference_results):
+    r"""
+    Overlays inferred bounding boxes on original image and writes a copy to image_dir
+
+    :param original_image: numpy array of the original image
+    :param inference_results: JSON object containing inference results
+    """
+
+    output_image = original_image
+    height, width, channels = output_image.shape
+    scale_factor_x = width / config_utils.WIDTH
+    scale_factor_y = height / config_utils.HEIGHT
+
+    config_utils.logger.info(inference_results)
+
+    for result in inference_results:
+        boundaries = fromstring(result["Boundaries"][1:-1], dtype=float, sep=" ")
+        box_label = "{} : {}".format(result["Label"], result["Score"])
+        x_min = floor(boundaries[0] * scale_factor_x)
+        y_min = floor(boundaries[1] * scale_factor_y)
+        x_max = floor(boundaries[2] * scale_factor_x)
+        y_max = floor(boundaries[3] * scale_factor_y)
+
+        # Draw bounding box
+        rectangle(
+            img=output_image,
+            pt1=(x_min, y_min),
+            pt2=(x_max, y_max),
+            color=(0, 0, 0),
+            thickness=2,
+        )
+
+        # Draw black text outline
+        putText(
+            img=output_image,
+            text=box_label,
+            org=(
+                x_min,
+                int(y_min - (0.02 * height)),
+            ),  # Draw text above top left of box
+            fontFace=FONT_HERSHEY_SIMPLEX,
+            fontScale=0.6,
+            color=(255, 255, 255),
+            thickness=3,
+        )
+        # Draw white text
+        putText(
+            img=output_image,
+            text=box_label,
+            org=(
+                x_min,
+                int(y_min - (0.02 * height)),
+            ),  # Draw text above top left of box
+            fontFace=FONT_HERSHEY_SIMPLEX,
+            fontScale=0.6,
+            color=(0, 0, 0),
+            thickness=2,
+        )
+
+    output_path = path.join(
+        config_utils.BOUNDED_OUTPUT_DIR, config_utils.DEFAULT_BOUNDED_OUTPUT_IMAGE_NAME
+    )
+    imwrite(output_path, output_image, [IMWRITE_JPEG_QUALITY, 100])
+    config_utils.logger.info("Output image with bounding boxes to {}".format(output_path))
+
+
+def predict(image_data):
+    r"""
+    Performs object detection and predicts using the model.
+
+    :param image_data: numpy array of the resized image passed in for inference.
+    :return: JSON object of inference results
+    """
+    PAYLOAD = {}
+    PAYLOAD["timestamp"] = str(datetime.now())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by passing `tzinfo` as an argument to the method.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)